src/mbgl/util/work_request.hpp

```diff
@@ -10,9 +10,10 @@ namespace mbgl {
 class WorkRequest : public util::noncopyable {
 public:
     using Future = std::future<void>;
+    using JoinedFlag = std::shared_ptr<std::atomic<bool>>;
```

Comment:
It appears that a `std::atomic_flag` can be initialized [only with the statement `std::atomic_flag v = ATOMIC_FLAG_INIT;`](http://en.cppreference.com/w/cpp/atomic/ATOMIC_FLAG_INIT), and we need to initialize the flag in a shared pointer, so I don't think it will work in this case.